extensions/blocks/tiled-gallery/tiled-gallery.php

```diff
@@ -26,14 +28,88 @@ class_exists( 'Jetpack_Photon' ) && Jetpack::is_module_active( 'photon' )
 	 *
 	 * @return string
 	 */
-	function jetpack_tiled_gallery_load_block_assets( $attr, $content ) {
+	public static function render( $attr, $content ) {
 		$dependencies = array(
 			'lodash',
 			'wp-i18n',
 			'wp-token-list',
 		);
 		Jetpack_Gutenberg::load_assets_as_required( 'tiled-gallery', $dependencies );
 
+		$is_squareish_layout = self::is_squareish_layout( $attr );
+		$srcset_step         = 300;
+
+		if ( function_exists( 'jetpack_photon_url' )
+			&& preg_match_all( '/<img [^>]+>/', $content, $images )
+		) {
+			/**
+			 * This block processes all of the images that are found and builds $find and $replace.
+			 *
+			 * The original img is added to the $find array and the replacement is made and added
+			 * to the $replace array. This is so that the same find and replace operations can be
+			 * made on the entire $content.
+			 */
+			$find    = array();
+			$replace = array();
+
+			foreach ( $images[0] as $image_html ) {
+				if (
+					preg_match( '/data-width="([0-9]+)"/', $image_html, $img_height )
+					&& preg_match( '/data-height="([0-9]+)"/', $image_html, $img_width )
+					&& preg_match( '/src="([^"]+)"/', $image_html, $img_src )
+				) {
+					$orig_width  = $img_width[1];
+					$orig_height = $img_height[1];
+
+					// `src`s are already "photonized". Drop their parameters so we can add our own.
+					$src_parts = explode( '?', $img_src[1], 2 );
```

Comment:
Would checking for hostname be a more robust indication of Photon, rather than just checking for URL params?

With help of [`parse-url`](https://secure.php.net/manual/en/function.parse-url.php) or something similar to what [Photon functions do](https://github.com/Automattic/jetpack/blob/f2bcdc78dd3ba56b9b07bde603f4d2005464ded6/functions.photon.php#L107-L117).

Reply:
A couple of givens:

- We're serializing photon-like (photon or files.wordpress.com) URLs from the editor already.
- Photon can't operate on image sources that have query parameters.

If we accept these two, then dropping the query string seems reasonable.